test/integration/clusterio.js

```diff
@@ -1,14 +1,39 @@
+/* eslint-disable no-console */
 "use strict";
 const assert = require("assert").strict;
 const fs = require("fs-extra");
 const path = require("path");
+const events = require("events");
+
 
 const libLink = require("@clusterio/lib/link");
 const libUsers = require("@clusterio/lib/users");
+const server = require("@clusterio/lib/factorio/server");
+const child_process = require("child_process");
 
 const { slowTest, get, exec, execCtl, sendRcon, getControl, instancesDir } = require("./index");
 
 
+function spawn(name, cmd, waitFor) {
+	return new Promise((resolve, reject) => {
+		console.log(cmd);
+		let parts = cmd.split(" ");
+		let process = child_process.spawn(parts[0], parts.slice(1), { cwd: path.join("temp", "test") });
+		let stdout = new server._LineSplitter((line) => {
+			line = line.toString("utf8");
+			if (waitFor.test(line)) {
+				resolve(process);
+			}
+			console.log(name, line);
+		});
+		let stderr = new server._LineSplitter((line) => { console.log(name, line.toString("utf8")); });
+		process.stdout.on("data", chunk => { stdout.data(chunk); });
+		process.stdout.on("close", () => { stdout.end(); });
+		process.stderr.on("data", chunk => { stderr.data(chunk); });
+		process.stderr.on("close", () => { stderr.end(); });
+	});
+}
+
```

Comment:
Do not copy and paste code like this. Just import it from `./index`. 😞

Reply:
yeah that would be better that also fixses the console.log thing kinda forgot i had this